pkgs/applications/editors/bluej/default.nix

```diff
@@ -11,26 +12,52 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-KhNhJ2xsw1g2yemwP6NQmJvk4cxZAQQNPEUBuLso5qM=";
   };
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ wrapGAppsHook ];
+  buildInputs = [ glib ];
 
-  unpackPhase = ''
-    ar xf $src
-    tar xf data.tar.xz
+  sourceRoot = ".";
+  preUnpack = ''
+    unpackCmdHooks+=(_tryDebData)
+    _tryDebData() {
+      if ! [[ "$1" =~ \.deb$ ]]; then return 1; fi
+      ar xf "$1"
+      if ! [[ -e data.tar.xz ]]; then return 1; fi
+      unpackFile data.tar.xz
```

Comment:
Same here. We know we're going to be unpack the deb and the containing data.tar.xz, so why try to make this generic?

Reply:
This allows for overriding `src` without also having to override unpacking.

Reply:
This can be simplified with `dpkg-deb -x`